Gridy/Controller/StartViewController.swift

```diff
@@ -18,7 +19,148 @@ class StartViewController: UIViewController {
+        case .notDetermined:
+            AVCaptureDevice.requestAccess(for: AVMediaType.video, completionHandler: { (granted) in
+                if granted {
+                    self.presentImagePicker(sourceType: sourceType)
+                    print ("Access authorized.")
+                } else {
+                    self.troubleAlert(message: noPermissionMessage)
+                }
+            })
+        case .authorized:
+            self.presentImagePicker(sourceType: sourceType)
+            print ("Access authorized.")
+        case .denied, .restricted:
+            self.troubleAlert(message: noPermissionMessage)
+        default:
+            self.troubleAlert(message: "Something went terribly wrong. Gridy is unable to access your photo library at this time.")
+        }
+    }
+    
+    /// Ask the user the authorization to use the photo library.
+    /// User can redirect to Settings if authorization not granted.
+    /// Call Image Picker if authorization has already been granted
+    ///
+    /// - Parameters:
+    ///   - status: Authorization status to use PhotoLibrary
+    ///   - noPermissionMessage: Message to display if authorization not granted
+    func actionAccordingTo(status: PHAuthorizationStatus , noPermissionMessage: String?) {
+        let sourceType = UIImagePickerController.SourceType.photoLibrary
+        switch status {
+        case .notDetermined:
+            PHPhotoLibrary.requestAuthorization({ (newStatus) in
+                if newStatus == .authorized {
+                    self.presentImagePicker(sourceType: sourceType)
+                    print ("Access authorized.")
+                }
+                else {
+                    self.troubleAlert(message: noPermissionMessage)
+                }
+            })
+        case .authorized:
+            self.presentImagePicker(sourceType: sourceType)
+            print ("Access authorized.")
+        case .denied, .restricted:
+            self.troubleAlert(message: noPermissionMessage)
+        default:
+            self.troubleAlert(message: "Something went terribly wrong. Gridy is unable to access your photo library at this time.")
+        }
+    }
 
-
+    /// Present image picker
+    /// - Parameter sourceType: Camera or photo library
+    func presentImagePicker(sourceType: UIImagePickerController.SourceType) {
+        let imagePicker = UIImagePickerController()
+        imagePicker.delegate = self
+        imagePicker.sourceType = sourceType
+        present(imagePicker, animated: true, completion: nil)
+    }
+    
+    /// Present trouble alert view
+    /// - Parameter message: Message for the user
+    func troubleAlert(message: String?) {
+            let alertController = UIAlertController(title: "Oops...", message: message, preferredStyle: .alert)
+            let OKAction = UIAlertAction(title: "Got it!", style: .cancel)
+            alertController.addAction(OKAction)
+            present(alertController, animated: true)
+    }
```

Comment:
It seems you have an indentation issue here

Reply:
yes ;) fixed